tools/shared.py

```diff
@@ -395,6 +394,11 @@ def check_sanity(force=False):
   if DEBUG:
     force = True
 
+  # Always check the llvm version since it only generates a warning.
+  # The version information is cached so `generate_sanity` below
+  # will used the cached version.
+  check_llvm_version()
```

Comment:
It seems like doing this all the time will cause us to an extra subprocess invocation each time emcc is run? That is, before we did 0 calls to `clang --version` on most emcc calls, and now we'll do exactly 1 in each emcc call?

Reply:
No, its still exactly 1 call on every invocation.   See the comment above.  

As part of the `expected = generate_sanity()` call just below we have to call `get_clang_version`.  This change just means that call `clang --version` happens slightly earlier.   The result of `get_clang_version` is always cached so there is never more than one call per process.

Reply:
I see.. unfortunate but seems like an existing issue.

Still, this does cause more work in `FROZEN_CACHE` and `EM_IGNORE_SANITY` modes? Aside from work, logically I think we shouldn't be checking stuff there?

Reply:
The `EM_IGNORE_SANITY` already unconditionally runs `perform_sanity_checks` where this function was moved form.

The frozen-ness of the cache also seems somewhat unrelated to weather we should be checking the llvm version or not. 
  I could move it to later.. but that would would have strange behavior the setting your cache to frozen also happens to suppress llvm version checking.

Reply:
> I see.. unfortunate but seems like an existing issue.

Do you think there is some way we can avoid this?  We check the sanity file on every invocation and I can't how to verify the sanity file is correct without checking the llvm version.   Perhaps we should just read the llvm path out of the sanity file and compare it to the one we have been configured with?  But then we would not catch the case where the llvm version was updated in-place (which happens in most cases that I know of).

Reply:
OK I came up with a version of this change that is less intrusive.